### PR TITLE
Translate llvm.loop.unroll.full metadata

### PR DESCRIPTION
**Pull request by MrSidims**
It can be generated via #pragma clang unroll(full) pragma. llvm.loop.unroll.full means attempt to do full unroll of the loop and disable the unrolling if the trip count is not known at compile time.

Unroll mask to which it was previously mapped doesn't match the description.

The way the patch represents it in SPIR-V is:
 Unroll mask + PartialCount mask with '1' parameter

This patch also removes some overtesting for unroll metadata.

Signed-off-by: Sidorov, Dmitry <dmitry.sidorov@intel.com>

### CODE REVIEW

File: test/transcoding/LoopUnroll.ll
Comment (MrSidims) [reply test/transcoding/LoopUnroll.ll:43]: I was compiling with O2 and with 'full' unroll the loop was being unrolled. Increased number of loop iterations to 1024 leaving 'full' pragma.

File: test/DebugInfo/DebugUnstructuredControlFlow.cl
Comment (MrSidims) [reply test/DebugInfo/DebugUnstructuredControlFlow.cl]: restored this test

**Review by bader — APPROVED** [review]
LGTM, but I'd like to get another approval from @svenvh before merging.

**Review by svenvh — APPROVED** [review]

**Comment by MrSidims** [comment]
Test failure is fixed in https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1672

**Pull request merged**
